scheduler/pkg/kafka/gateway/client.go

```diff
@@ -153,14 +153,19 @@ func (kc *KafkaSchedulerClient) SubscribeModelEvents() error {
 
 		logger.Infof("Received event name %s version %d state %s", event.ModelName, latestVersionStatus.Version, latestVersionStatus.State.State.String())
 
-		switch latestVersionStatus.State.State {
-		case scheduler.ModelStatus_ModelAvailable:
+		// if there are available replicas then we add the consumer for the model
+		// note that this will also get triggered if the model is already added and in this case it is a no-op
```

Comment:
Just checking my understanding: this part also documents previous behaviour, right? As in, we weren't differentiating between already added models previously either.

Reply:
yes